db/db_impl/db_impl_write.cc

```diff
@@ -1584,9 +1588,17 @@ Status DBImpl::SwitchWAL(WriteContext* write_context) {
 Status DBImpl::HandleWriteBufferManagerFlush(WriteContext* write_context) {
   mutex_.AssertHeld();
   assert(write_context != nullptr);
-  assert(immutable_db_options_.flush_switch->IsFlushOn());
   Status status;
 
+  // If any CF has flush disabled, we shouldn't handle write buffer full
+  if (auto cfd_flush_disabled = GetAnyCFWithFlushDisabled()) {
+    ROCKS_LOG_INFO(immutable_db_options_.info_log,
+                   "HandleWriteBufferManagerFlush is not triggered due to "
+                   "flush disabled for CF: %d",
+                   cfd_flush_disabled->GetID());
+    return Status::OK();
+  }
```

Comment:
You don't need this if you follow the suggestion above about only disabling auto flush.

Reply:
You can argue it is, but the caller controls it fully (through `WriteBufferManager`), so they already have the tools to stop flushes triggered that way.

Reply:
Got it. yeah. we can always set `db_write_buffer_size` to 0 to disable it (which is the case for us right now)